eng/packages/http-client-csharp-mgmt/emitter/test/resource-detection.test.ts

```diff
@@ -378,6 +391,28 @@ interface CurrentEmployees {
       currentMetadata.resourceName,
       "CurrentEmployee"
     );
+    
+    // Validate using resolveArmResources API
+    const resolvedSchema = resolveArmResources(program, sdkContext);
+    ok(resolvedSchema);
+    ok(resolvedSchema.resources);
+    
+    // Verify both singleton resources exist (if resolveArmResources includes them)
+    const resolvedEmployee = resolvedSchema.resources.find(
+      (r) => r.metadata.resourceType === "Microsoft.ContosoProviderHub/employees"
+    );
+    if (resolvedEmployee) {
+      strictEqual(resolvedEmployee.metadata.resourceIdPattern, metadata.resourceIdPattern);
+      strictEqual(resolvedEmployee.metadata.singletonResourceName, "default");
+    }
+    
+    const resolvedCurrentEmployee = resolvedSchema.resources.find(
+      (r) => r.metadata.resourceType === "Microsoft.ContosoProviderHub/currentEmployees"
+    );
+    if (resolvedCurrentEmployee) {
+      strictEqual(resolvedCurrentEmployee.metadata.resourceIdPattern, currentMetadata.resourceIdPattern);
+      strictEqual(resolvedCurrentEmployee.metadata.singletonResourceName, "current");
+    }
```

Comment:
So let us apply the same pattern in the above test case into all test cases in this file.